rcl_action/src/rcl_action/action_server.c

```diff
@@ -395,6 +421,66 @@ rcl_action_accept_new_goal(
   return goal_handles[num_goal_handles];
 }
 
+// Implementation only
+static rcl_ret_t
+_recalculate_expire_timer(
+  rcl_timer_t * expire_timer,
+  const int64_t timeout,
+  rcl_action_goal_handle_t ** goal_handles,
+  size_t num_goal_handles,
+  rcl_clock_t * clock)
+{
+  size_t num_inactive_goals = 0u;
+  int64_t minimum_period = 0;
+
+  // Get current time (nanosec)
+  int64_t current_time;
+  rcl_ret_t ret = rcl_clock_get_now(clock, &current_time);
+  if (RCL_RET_OK != ret) {
+    return RCL_RET_ERROR;
```

Comment:
```suggestion
    return ret;
```

return `ret` instead?

Reply:
I'm not sure if it's convention to hide the error code to match the documentation or return the more specific one. It's inconsistent in `rcl`, but `rcl_action` is consistent about doing the former. :man_shrugging: